### PR TITLE
New Resource CDDC-CddcDedicatedHostGroup and its DataSource

### PR DESCRIPTION
**Pull request by Wanghx0991**

### CODE REVIEW

File: alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	allConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudCddcDedicatedHostGroupsDataSourceName(rand, map[string]string{
+			"ids":            `["${alicloud_cddc_dedicated_host_group.default.id}"]`,
+			//"image_category": `"AliLinux"`,
+			"engine": `"MySQL"`,
+		}),
+		fakeConfig: testAccCheckAlicloudCddcDedicatedHostGroupsDataSourceName(rand, map[string]string{
+			"ids":            `["${alicloud_cddc_dedicated_host_group.default.id}_fake"]`,
+			//"image_category": `"AliLinux_fake"`,
+			"engine": `"MySQL_fake"`,
```
[the remark this answers is not in the document]
Comment (Wanghx0991) [reply alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go:31]: done

File: alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		}),
+	}
+	engineConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudCddcDedicatedHostGroupsDataSourceName(rand, map[string]string{
+			"engine": `"MySQL"`,
+		}),
+		fakeConfig: testAccCheckAlicloudCddcDedicatedHostGroupsDataSourceName(rand, map[string]string{
+			"engine": `"MySQL_fake"`,
+		}),
+	}
```
[the remark this answers is not in the document]
Comment (Wanghx0991) [reply alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go]: done

File: alicloud/resource_alicloud_cddc_dedicated_host_group.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+	}
+	d.Set("dedicated_host_group_desc", object["DedicatedHostGroupDesc"])
+	if v, ok := object["DiskAllocationRatio"]; ok && fmt.Sprint(v) != "0" {
+		d.Set("disk_allocation_ratio", formatInt(v))
+	}
+
+	engine := object["Engine"].(string)
+	switch engine {
+	case "mysql":
+		engine = "MySQL"
```
[the remark this answers is not in the document]
Comment (Wanghx0991) [reply alicloud/resource_alicloud_cddc_dedicated_host_group.go]: done

File: alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	for k, v := range attrMap {
+		pairs = append(pairs, k+" = "+v)
+	}
+	config := fmt.Sprintf(`
+
+variable "name" {	
+	default = "tf-testAccName-%d"
+}
+resource "alicloud_vpc" "vpc" {
+  vpc_name   = "tf_test_foo"
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_cddc_dedicated_host_groups_test.go:75]: 一样的问题

File: alicloud/resource_alicloud_cddc_dedicated_host_group_test.go [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+}
+var AlicloudCDDCDedicatedHostGroupMap0 = map[string]string{
+	"engine": CHECKSET,
+	"vpc_id": CHECKSET,
+}
+
+func AlicloudCDDCDedicatedHostGroupBasicDependence0(name string) string {
+	return fmt.Sprintf(` 
+resource "alicloud_vpc" "vpc" {
+  vpc_name   = "tf_test_foo"
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_cddc_dedicated_host_group_test.go:640]: 一样的

File: website/alicloud.erb [hunk 6]
```diff
@@ -391,6 +413,28 @@
                   </ul>
                 </li>
 
+                <li>
+                  <a href="#">CDDC</a>
+                  <ul class="nav">
+                      <li>
+                        <a href="#">Data Sources</a>
+                        <ul class="nav nav-auto-expand"
```
Comment (xiaozhu36) [inline website/alicloud.erb:421]: 怎么有两个

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**